lldb/source/API/SBPlatform.cpp

```diff
@@ -574,6 +576,29 @@ SBError SBPlatform::Launch(SBLaunchInfo &launch_info) {
   });
 }
 
+SBProcess SBPlatform::Attach(SBAttachInfo &attach_info,
+                             const SBDebugger &debugger, SBTarget &target,
```

Comment:
I'd like to see the `attach_info` parameter marked const if possible, but it looks like `SBAttachInfo::ref` is not const. Maybe we can add an overload?

I think that you can do `const SBTarget &target` without any further changes because `SBTarget::GetSP` is marked `const`.

Reply:
- `attach_info` cannot be const because `Platform::Attach` also takes a non-cost `AttachInfo`. There's at least one place (`PlatformPOSIX::Attach`) that modifies the `attach_info`. 
- `target` is an out parameter so making that const would be confusing.